File: Python/Lv.2/6.py
//짝지어 제거하기
def solution(s):
    answer = -1
    
    ss = []; 
    i = 0;
    
    ss.append(s[i]);
    
    while(1):
        i += 1;
        
        if(i > len(s) - 1):
            break;
        
        if(len(ss) != 0):
            if(ss[-1] == s[i]):
                ss.pop();
            else:
                ss.append(s[i]);
        else:
            ss.append(s[i]);
    
    
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    if(len(ss) == 0):
        answer = 1;
    else:
        answer = 0;
    
    return answer

File: C/Lv.2/6.c
//n^2 배열 자르기
//idea (x, y)일 때 max(x, y) + 1
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n, long long left, long long right) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int* answer = (int*)malloc(sizeof(int) * (right - left + 1));
    int tm1, tm2, tp1, tp2;
    
    tm1 = left / n;
    tp1 = left % n;

    for(int i = 0; i < right - left + 1; i++) {
        if(tm1 > tp1)
            answer[i] = tm1 + 1;
        else
            answer[i] = tp1 + 1;
        tp1 += 1;
        if(tp1 == n){
            tp1 = 0;
            tm1 += 1;
        }
    }
    
    return answer;
}